Android/Yash Raj Gupta/Intermediate/ToDoApp/app/src/main/java/com/example/todoapp/MainActivity.java

```diff
@@ -0,0 +1,101 @@
+package com.example.todoapp;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.example.todoapp.Adapter.TaskAdapter;
+import com.example.todoapp.DocumentId.TaskModel;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainActivity extends AppCompatActivity {
+
+    FirebaseFirestore db = FirebaseFirestore.getInstance();
```

Comment:
Update task operation is not implemented (only insert and delete operation functionality is present.) Please check. Suggestion : Can add an extra button for update for every list item in the recylcerview, when user clicks it, load the list item in the edit text and update the entry in the DB with new changes after user clicks on add button again.